### PR TITLE
feat: support for `_routes.json` in the root dir

### PR DESCRIPTION
**Pull request by james-elicx**
This PR does the following:
- Copies the entries from an `_routes.json` file found in the root directory to our generated version.

### CODE REVIEW

File: packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts [hunk 1]
```diff
@@ -44,19 +54,39 @@ ${Object.entries(nextStaticHeaders)
...
 	try {
 		await writeFile(
 			join(outputDir, '_routes.json'),
 			JSON.stringify({
 				version: 1,
 				description: `Built with @cloudflare/next-on-pages@${nextOnPagesVersion}.`,
-				include: ['/*'],
-				exclude: [`${nextStaticPath}/*`],
+				include: includeEntries,
+				exclude: [...excludeEntries],
```
Comment (dario-piotrowicz) [inline packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts:88]: why the spreading? 😮 

```suggestion
				exclude: excludeEntries,
```

**Review by dario-piotrowicz — APPROVED** [review]
Looks good to me 🙂👍 

Up to you if you also want to add this somewhere in our documentation, let me know if you want to do that otherwise we can merge

**Comment by james-elicx** [comment]
> Looks good to me 🙂👍
> 
> Up to you if you also want to add this somewhere in our documentation, let me know if you want to do that otherwise we can merge

little easter egg perhaps

File: packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts [hunk 2]
```diff
@@ -44,19 +54,39 @@ ${Object.entries(nextStaticHeaders)
...
+	if (existingFile && existingFile.version !== 1) {
+		cliError(
+			`Found _routes.json with version ${existingFile.version}, but only version 1 is supported.`,
+		);
+		process.exit(1);
+	}
+
+	const { include = [], exclude = [] } = existingFile ?? {};
+	const includeEntries = include.length > 0 ? include : ['/*'];
+	const excludeEntries = new Set([`${nextStaticPath}/*`, ...(exclude ?? [])]);
```
Comment (james-elicx) [inline packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts:79]:
```suggestion
	const excludeEntries = new Set([`${nextStaticPath}/*`, ...exclude]);
```

**Comment by asimpletune** [comment]
Has this changed? I noticed more recently that the documentation says _routes.json belongs in the build output's directory. Based on my local testing this also appears to be the case.

**Pull request merged**